src/algorithm.rs

```diff
@@ -668,6 +716,79 @@ impl Bft {
         PRECOMMIT_ON_NOTHING
     }
 
+    #[cfg(feature = "verify_req")]
+    fn check_precommit_count(&mut self) -> i8 {
+        if let Some(precommit_set) =
+            self.votes
+                .get_voteset(self.height, self.round, Step::Precommit)
+        {
+            let mut tv = if self.cal_all_vote(precommit_set.count) {
+                Duration::new(0, 0)
+            } else {
+                self.params.timer.get_precommit()
+            };
+            if !self.cal_above_threshold(precommit_set.count) {
+                return PRECOMMIT_BELOW_TWO_THIRDS;
+            }
+
+            info!(
+                "Receive over 2/3 precommit at height {:?}, round {:?}",
+                self.height, self.round
+            );
+            let mut pending_flag = false;
+            let mut clean_flag = false;
+
+            for (hash, count) in &precommit_set.votes_by_proposal {
+                if self.cal_above_threshold(*count) {
+                    if hash.is_empty() {
+                        info!("Reach nil consensus, goto next round {:?}", self.round + 1);
+                        return PRECOMMIT_ON_NIL;
+                    } else {
+                        self.set_polc(&hash, &precommit_set, Step::Precommit);
+                        if let Some(is_pass) = self.verify_result.get(hash) {
+                            // check verify result of the proposal
+                            if !is_pass {
+                                // if verified fail, clean PoLC and proposal
+                                info!("The verify result of proposal {:?} is disapproved.", &hash);
+                                clean_flag = true;
+                                self.proposal = None;
```

Comment:
This is redundant because clean flag will clean it again.

Reply:
This is not redundant, if not clean PoLC here, it will take the iffy lock to go to next round.